### PR TITLE
Merging loops and 2 bugfixes

### PR DESCRIPTION
**Pull request by DomT602**
Resolves #. 

#### Changes proposed in this pull request: 
- Merges the 3 server loops (cleanup, federal update and dealers) into one
- Also addresses 2 bugs (gang funds and undefined _query)

- [x] I have tested my changes and corrected any errors found

### CODE REVIEW

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 1]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
 };
+
+//Array format: [parameters,function,delayTime]
+private _routines = [
+    [[],_fnc_fedDealerUpdate,1800],
+    [[],_fnc_cleanVehicles,3600]
+];
+
+{
+    _x pushBack diag_tickTime;
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf]:
```suggestion
    _x pushBack (diag_tickTime + (_x # 2));
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 2]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+} forEach _routines;
+
+for "_i" from 0 to 1 step 0 do {
+    {
+        _x params ["_params","_function","_delay","_timeToRun"];
+        if (diag_tickTime > _timeToRun) then {
+            _x set [2,_timeToRun + _delay];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf]:
```suggestion
            _x set [2, diag_tickTime + _delay];
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 3]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+            _params call _function;
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf]: Should we move this above the set? Might take awhile to run (theoretical)

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 4]
```diff
@@ -46,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+
+for "_i" from 0 to 1 step 0 do {
+    {
+        _x params ["_params","_function","_delay","_timeToRun"];
+        if (diag_tickTime > _timeToRun) then {
+            _x set [2,_timeToRun + _delay];
+            _params call _function;
+        };
+    } forEach _routines;
+    uiSleep 1e-6;
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf]: Ditto with other comments

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 5]
```diff
@@ -3,23 +3,28 @@
...
+    Server-side cleanup script on vehicles, dealers and fed reserve.
 */
 private _saveFuel = LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1;
 private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
 
-for "_i" from 0 to 1 step 0 do {
-    uiSleep (60 * 60);
+private _fnc_fedDealerUpdate = {
+    {
+        _x setVariable ["sellers",[],true];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:14]: Perhaps it's time to save ourselves some time helping people on the Discord, and check if they're nil first

```suggestion
        private _dealer = missionNamespace getVariable [_x, nil];
        if (!isNil "_dealer") then {
            _dealer setVariable ["sellers", [], true];
        };
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 6]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
         true
     } count (allMissionObjects "Thing");
 };
+
+//Array format: [parameters,function,delayTime]
+private _routines = [
+    [[],_fnc_fedDealerUpdate,1800],
+    [[],_fnc_cleanVehicles,3600]
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:61]:
```suggestion
    [[], _fnc_cleanVehicles, 3600]
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 7]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
         true
     } count (allMissionObjects "Thing");
 };
+
+//Array format: [parameters,function,delayTime]
+private _routines = [
+    [[],_fnc_fedDealerUpdate,1800],
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:60]:
```suggestion
    [[], _fnc_fedDealerUpdate, 1800],
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 8]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+{
+    _x pushBack (diag_tickTime + (_x # 2));
+} forEach _routines;
+
+for "_i" from 0 to 1 step 0 do {
+    {
+        _x params ["_params","_function","_delay","_timeToRun"];
+        if (diag_tickTime > _timeToRun) then {
+            _params call _function;
+            _x set [2,diag_tickTime + _delay];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:73]:
```suggestion
            _x set [2, diag_tickTime + _delay];
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 9]
```diff
@@ -46,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
         true
     } count (allMissionObjects "Thing");
 };
+
+//Array format: [parameters,function,delayTime]
+private _routines = [
+    [[],_fnc_fedDealerUpdate,1800],
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:60]:
```suggestion
    [[], _fnc_fedDealerUpdate, 1800],
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 10]
```diff
@@ -46,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+    [[],_fnc_cleanVehicles,3600]
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:61]:
```suggestion
    [[], _fnc_cleanVehicles, 3600]
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 11]
```diff
@@ -46,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+{
+    _x pushBack (diag_tickTime + (_x # 2));
+} forEach _routines;
+
+for "_i" from 0 to 1 step 0 do {
+    {
+        _x params ["_params","_function","_delay","_timeToRun"];
+        if (diag_tickTime > _timeToRun) then {
+            _params call _function;
+            _x set [2,diag_tickTime + _delay];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:73]:
```suggestion
            _x set [2, diag_tickTime + _delay];
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 12]
```diff
@@ -46,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+    [[],_fnc_cleanVehicles,3600]
+];
+
+{
+    _x pushBack (diag_tickTime + (_x # 2));
+} forEach _routines;
+
+for "_i" from 0 to 1 step 0 do {
+    {
+        _x params ["_params","_function","_delay","_timeToRun"];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:70]:
```suggestion
        _x params ["_params", "_function", "_delay", "_timeToRun"];
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 13]
```diff
@@ -4,19 +4,27 @@
...
 
-for "_i" from 0 to 1 step 0 do {
-    uiSleep (60 * 60);
+private _fnc_fedDealerUpdate = {
+    {
+        _x setVariable ["sellers",[],true];
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
+
+    private _funds = fed_bank getVariable ["safe",0];
+    fed_bank setVariable ["safe",round(_funds+((count playableUnits)/2)),true];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:18]:
```suggestion
    fed_bank setVariable ["safe", round(_funds + ((count playableUnits)/2)), true];
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 14]
```diff
@@ -49,3 +54,24 @@ for "_i" from 0 to 1 step 0 do {
...
+    [[],_fnc_cleanVehicles,3600]
+];
+
+{
+    _x pushBack (diag_tickTime + (_x # 2));
+} forEach _routines;
+
+for "_i" from 0 to 1 step 0 do {
+    {
+        _x params ["_params","_function","_delay","_timeToRun"];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:70]:
```suggestion
        _x params ["_params", "_function", "_delay", "_timeToRun"];
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 15]
```diff
@@ -3,23 +3,28 @@
...
 private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
 
-for "_i" from 0 to 1 step 0 do {
-    uiSleep (60 * 60);
+private _fnc_fedDealerUpdate = {
+    {
+        _x setVariable ["sellers",[],true];
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
+
+    private _funds = fed_bank getVariable ["safe",0];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:17]:
```suggestion
    private _funds = fed_bank getVariable ["safe", 0];
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 16]
```diff
@@ -3,23 +3,28 @@
...
+    fed_bank setVariable ["safe",round(_funds+((count playableUnits)/2)),true];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:18]:
```suggestion
    fed_bank setVariable ["safe", round(_funds+((count playableUnits)/2)), true];
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 17]
```diff
@@ -4,19 +4,27 @@
...
 private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
 
-for "_i" from 0 to 1 step 0 do {
-    uiSleep (60 * 60);
+private _fnc_fedDealerUpdate = {
+    {
+        _x setVariable ["sellers",[],true];
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
+
+    private _funds = fed_bank getVariable ["safe",0];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:17]:
```suggestion
    private _funds = fed_bank getVariable ["safe", 0];
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 18]
```diff
@@ -3,23 +3,28 @@
...
 */
 private _saveFuel = LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1;
 private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
 
-for "_i" from 0 to 1 step 0 do {
-    uiSleep (60 * 60);
+private _fnc_fedDealerUpdate = {
+    {
+        _x setVariable ["sellers",[],true];
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:15]:
```suggestion
    } forEach ["Dealer_1", "Dealer_2", "Dealer_3"];
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 19]
```diff
@@ -4,19 +4,27 @@
...
+    Server-side cleanup script on vehicles, dealers and fed reserve.
 */
 private _saveFuel = LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1;
 private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
 
-for "_i" from 0 to 1 step 0 do {
-    uiSleep (60 * 60);
+private _fnc_fedDealerUpdate = {
+    {
+        _x setVariable ["sellers",[],true];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:14]:
```suggestion
        private _dealer = missionNamespace getVariable [_x, nil];
        if (!isNil "_dealer") then {
            _dealer setVariable ["sellers", [], true];
        };
```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 20]
```diff
@@ -4,19 +4,27 @@
...
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:15]:
```suggestion
    } forEach ["Dealer_1", "Dealer_2", "Dealer_3"];
```

File: life_server/Functions/Systems/fn_cleanup.sqf (hunk 19 above)
Comment (BlackFisch) [reply life_server/Functions/Systems/fn_cleanup.sqf:14]: if you do `missionNamespace getVariable [_x, nil];` you could also just go with `missionNamespace getVariable _x;`
default value makes only sense it its not nil

Comment (BoGuu) [reply life_server/Functions/Systems/fn_cleanup.sqf:14]: potayto, potahto :)

Comment (BlackFisch) [reply life_server/Functions/Systems/fn_cleanup.sqf:14]: just unnecessary code in this case :D

Comment (BlackFisch) [inline life_server/Functions/Systems/fn_cleanup.sqf:14]:
```suggestion
        private _dealer = missionNamespace getVariable [_x, objNull];
        if (!isNull _dealer) then {
            _dealer setVariable ["sellers", [], true];
        };
```

File: life_hc/MySQL/General/fn_cleanup.sqf [hunk 21]
```diff
@@ -1,51 +1,80 @@
...
+private _saveFuel = LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1;
+private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
+
+private _fnc_fedDealerUpdate = {
+    {
+        private _dealer = missionNamespace getVariable [_x, objNull];
+        if !(isNull _dealer) then {
+            _x setVariable ["sellers", [], true];
+        };
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
```
Comment (BoGuu) [inline life_hc/MySQL/General/fn_cleanup.sqf:18]:
```suggestion
    } forEach ["Dealer_1", "Dealer_2", "Dealer_3"];

```

File: life_server/Functions/Systems/fn_cleanup.sqf [hunk 22]
```diff
@@ -1,48 +1,80 @@
...
+private _saveFuel = LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1;
+private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
+
+private _fnc_fedDealerUpdate = {
+    {
+        private _dealer = missionNamespace getVariable [_x, objNull];
+        if !(isNull _dealer) then {
+            _x setVariable ["sellers", [], true];
+        };
+    } forEach [Dealer_1,Dealer_2,Dealer_3];
```
Comment (BoGuu) [inline life_server/Functions/Systems/fn_cleanup.sqf:18]:
```suggestion
    } forEach ["Dealer_1", "Dealer_2", "Dealer_3"];
```

**Pull request merged**